tests/functional/test_report_only_no_errors.py

```diff
@@ -0,0 +1,319 @@
+# -*- coding: utf-8 -*-
+"""Report only submission
+
+Partners or redhat associates can publish their chart by submitting
+error-free report that was generated by chart-verifier.
+"""
+import os
+from dataclasses import dataclass
+from string import Template
+import tempfile
+import json
+import base64
+import git
+import pathlib
+from retrying import retry
+import logging
+import yaml
+from .utils import get_name_and_version_from_report
+
+import requests
+import pytest
+from pytest_bdd import (
+    given,
+    scenario,
+    then,
+    when,
+    parsers,
+)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+@retry(stop_max_delay=30_000)
+def get_run_id(headers, secrets):
+    r = requests.post(
+        f'https://api.github.com/repos/{secrets.test_repo}/pulls/{secrets.pr_number}', headers=headers, json={})
+    pr = json.loads(r.text)
+
+    r = requests.get(
+        f'https://api.github.com/repos/{secrets.test_repo}/actions/runs', headers=headers)
+    runs = json.loads(r.text)
+
+    for run in runs['workflow_runs']:
+        if run['head_sha'] == pr['head']['sha']:
+            return run['id']
+    else:
+        pytest.fail("Workflow for the submitted PR did not run.")
+
+
+@retry(stop_max_delay=60_000*2)
+def get_run_result(headers, secrets, run_id):
+    r = requests.get(
+        f'https://api.github.com/repos/{secrets.test_repo}/actions/runs/{run_id}', headers=headers)
+    run = json.loads(r.text)
+
+    if run['conclusion'] is None:
+        pytest.fail("Workflow conclusion is null.")
+
+    return run['conclusion']
+
+
+@pytest.fixture
+def secrets():
+    @dataclass
```

Comment:
you like data classes? I would have used a dictionary here, but I hear you.

Reply:
It was carried from the previous draft which i found was pretty neat.. because instead of `secrets['key_value']` we can do `secrets.key_value`

Reply:
K, lets roll with it.